pkg/parquetquery/iters.go

```diff
@@ -933,7 +937,7 @@ func (c *SyncIterator) makeResult(t RowNumber, v *pq.Value) *IteratorResult {
 	r := GetResult()
 	r.RowNumber = t
 	if c.selectAs != "" {
-		r.AppendValue(c.selectAs, v.Clone())
+		r.AppendValue(c.selectAs, c.interner.UnsafeClone(v))
```

Comment:
on a very high cardinality column will this be slower than just cloning?

should we compare the dictionary length to the page length and decide whether to intern or not?

Reply:
This is a nice suggestion. Added logic to skip UUIDs. I don't think it's worth going beyond that. The overhead of using the interner when the column has high cardinality is very little.